contracts/YearnCrvTwoTokens.sol

```diff
@@ -0,0 +1,122 @@
+ */
+contract YearnCrvTwoTokens is ITrigger {
+  // --- Tokens ---
+  // Token addresses
+  IERC20 internal immutable token0;
+  IERC20 internal immutable token1;
+
+  // Token indices in Curve pool arrays
+  uint256 internal constant token0Index = 0;
+  uint256 internal constant token1Index = 1;
+
+  // --- Tolerances ---
+  /// @dev Scale used to define percentages. Percentages are defined as tolerance / scale
+  uint256 public constant scale = 1000;
+
+  /// @dev In Yearn V2 vaults, the pricePerShare decreases immediately after a harvest, and typically ramps up over the
+  /// next six hours. Therefore we cannot simply check that the pricePerShare increases. Instead, we consider the vault
+  /// triggered if the pricePerShare drops by more than 50% from it's previous value. This is conservative, but
+  /// previous Yearn bugs resulted in pricePerShare drops of 0.5% – 10%, and were only temporary drops with users able
+  /// to be made whole. Therefore this trigger requires a large 50% drop to minimize false positives. The tolerance
+  /// is defined such that we trigger if: currentPricePerShare < lastPricePerShare * tolerance / 1000. This means
+  /// if you want to trigger after a 20% drop, you should set the tolerance to 1000 - 200 = 800
+  uint256 public constant vaultTol = scale - 500; // 50% drop, represented on a scale where 1000 = 100%
+
+  /// @dev Consider trigger toggled if Curve virtual price drops by more than this percentage.
+  uint256 public constant virtualPriceTol = scale - 500; // 50% drop
+
+  /// @dev Consider trigger toggled if Curve internal balances are lower than true balances by this percentage
+  uint256 public constant balanceTol = scale - 500; // 50% drop
+
+  // --- Trigger Data ---
+  /// @notice Yearn vault this trigger is for
+  IYVaultV2 public immutable vault;
+
+  /// @notice Curve tricrypto pool used as a strategy by `vault`
+  ICrvTricrypto public immutable curve;
+
+  /// @notice Last read pricePerShare
+  uint256 public lastPricePerShare;
+
+  /// @notice Last read curve virtual price
+  uint256 public lastVirtualPrice;
+
+  // --- Constructor ---
+
+  /**
+   * @param _vault Address of the Yearn V2 vault this trigger should protect
+   * @param _curve Address of the Curve Tricrypto pool uses by the above Yearn vault
+   * @param _token0 Address of token0 in the Curve pool
+   * @param _token1 Address of token1 in the Curve pool
+   * @dev For definitions of other constructor parameters, see ITrigger.sol
+   */
+  constructor(
+    string memory _name,
+    string memory _symbol,
+    string memory _description,
+    uint256[] memory _platformIds,
+    address _recipient,
+    address _vault,
+    address _curve,
+    IERC20 _token0,
+    IERC20 _token1
+  ) ITrigger(_name, _symbol, _description, _platformIds, _recipient) {
+    // Set trigger data
+    vault = IYVaultV2(_vault);
+    curve = ICrvTricrypto(_curve);
+    token0 = _token0;
```

Comment:
I'm guessing the vault / curve expose these token addresses as public params. If so, would it make sense to ask the addresses, rather than passing them into the constructor?

Reply:
Good call, done in 6efa9bcf15165324644e3f4bde8b461369bbce3d